src/net.md

```diff
@@ -1019,6 +1020,151 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-progress-with-range]: #ex-progress-with-range
+<a name="ex-progress-with-range"/>
+## Make a partial download with HTTP range headers
+
+[![reqwest-badge]][reqwest] [![cat-net-badge]][cat-net]
+
+Uses [`reqwest::Client::head`] to get the content-length and validate if the server sets the header
+[`reqwest::header::ContentRange`], required to confirm the support of partial downloads.
+
+If supported downloads the content using [`reqwest::get`], setting the [`reqwest::header::Range`]
+to do partial downloads in chunks of 100 bytes, between these writes basic progress messages.
+
+Range header, defined in [RFC7233][HTTP Range RFC7233].
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate reqwest;
+
+use std::io::{Read, Write};
+use std::fs::File;
+
+use reqwest::header::{ContentRange, ContentRangeSpec, Range};
+
+# error_chain! {
+#     foreign_links {
+#         Io(std::io::Error);
+#         Reqwest(reqwest::Error);
+#     }
+# }
+
+const BUFFER_SIZE: u64 = 100;
```

Comment:
the example is no_run anyway. we can go wild with the buffer and request size 
the biggest request handled by httpbin seams to be around 102400 but I did not check precisely. The buffer can be set to something like 10KB to still show partial content (this is still unreasonably small. we might have to add a comment that normally it would be bigger )